frontend/app/controllers/Info.scala

```diff
@@ -152,6 +152,14 @@ trait Info extends Controller {
     Ok(views.html.info.patron(pageInfo, pageImages))
   }
 
+  def offersAndCompetitions = CachedAction { implicit request =>
+    val results = GuardianContentService.offersAndCompetitionsContent.map(ContentItemOffer)
+      .filter(_.content.fields.map(_("membershipAccess")).isEmpty)
+      .filter(!_.content.webTitle.startsWith("EXPIRED"))
+      .filter(_.imgOpt.nonEmpty)
```

Comment:
Suboptimal code! By calling `filter` three times you are actually traversing the sequence three times. Replace with:

```scala

    val results = 
      GuardianContentService.offersAndCompetitionsContent.map(ContentItemOffer)
         .filter(offer => offer.content.fields.map(_("membershipAccess")).isEmpty &&
                          offer.content.webTitle.startsWith("EXPIRED") &&
                          offer.imgOpt.nonEmpty)

Also, why not move this into a method of `GuardianContentService` and keep the controller a bit more dry?
```